### DISCUSSION TITLE
Looking for API Endpoint/Report to Retrieve All Listings for Vendor Account

### DISCUSSION THREAD

**Discussion by egesenli** [amzn/selling-partner-api-models#3986] [Q&A]
I'm currently working on integrating a vendor account with the Amazon Vendor Central API. My goal is to retrieve all listings associated with a vendor account. However, I have encountered some difficulties and have not found any relevant information in the documentation.

The GET_MERCHANT_LISTINGS_ALL_DATA endpoint, which is available for sellers, does not seem to be applicable to vendors. I have tried using the GET_VENDOR_INVENTORY_REPORT to get the ASINs, but it does not provide the necessary details such as listing status, SKU, and other listing-related data.

Could you please advise if there is any API endpoint available for vendors that can provide a comprehensive list of all listings, including details like SKU, listing status, and other relevant information? SKU seems vital for also updating the listings via Feeds API or Catalog API.

If a direct solution is not available, I would greatly appreciate any alternative approaches or workarounds that can help achieve this objective.

Thank you for your assistance.

**Comment by chapmanjw** [comment]
@egesenli Today, you can use one of the vendor reports to identify which SKUs/ASINs are associated with the vendor and then use the Catalog Items API or Listings Items API to retrieve the data you're looking for.

With that said, we are currently working on a search operation for the Listings Items API that will enable you to search and paginate through all listings for a given seller or vendor. Stay tuned for announcements later this year.

**Reply by egesenli** [reply]
Hi @chapmanjw,

Thank you for your response.

I appreciate the information provided. However, my primary concern is obtaining the SKUs associated with the vendor listings. While I can retrieve ASINs using the GET_VENDOR_INVENTORY_REPORT and subsequently use the Catalog Items API or Listings Items API to fetch additional data, the SKUs remain crucial for updating listings via the Feeds API or Catalog API.

Could you please clarify if there is a way to directly obtain the SKUs of any existing Vendor Central API reports or endpoints?

Your assistance is greatly appreciated.
